### PR TITLE
[]: Replace ts-jest and jasmine2 with @swc/jest and jest-circus

### PR DESCRIPTION
**Pull request by samuelreichert**
## Checklist

-   Contains unit tests ✅ ❌
-   Contains breaking changes ✅ ❌
-   Compatible with: MX 7️⃣, 8️⃣, 9️⃣
-   Did you update version and changelog? ✅ ❌
-   PR title properly formatted (`[XX-000]: description`)? ✅ ❌

## This PR contains

-   [ ] Bug fix
-   [ ] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

### What

Swap the Jest transform and runner in `@mendix/pluggable-widgets-tools`'s shared test config.

**`test-config/jest.config.js`**
- Transform: `ts-jest` → `@swc/jest` (unified `(t|j)sx?` pattern replaces two separate rules)
- Runner: `jest-jasmine2` → `jest-circus/runner`
- Added: `testTimeout: 10000`

**`package.json`**
- Added deps: `@swc/core ^1.10.0`, `@swc/jest ^0.2.37`

### Why

**`ts-jest` is slow.** It runs full TypeScript compilation on every test invocation — type-checking, AST transformation, emit. `@swc/jest` is a Rust-native transpiler. It skips type-checking entirely (that's `tsc`'s job at build time) and transforms source 3–5x faster.

**`jest-jasmine2` is legacy.** Jest replaced it with `jest-circus` as the default runner in Jest 27. Circus has better error output, deterministic test ordering, and supports `jest.retryTimes()`. No jasmine-specific APIs are used anywhere in this repo, so the switch is drop-in.

**`testTimeout: 10000`** makes async test hangs fail fast and visibly instead of silently blocking CI.

### SWC config notes

| Option | Value | Reason |
|---|---|---|
| `jsc.target` | `es2019` | Matches the previous `ts-jest` `target: "ES2019"` |
| `jsc.parser` | `typescript`, `tsx: true`, `decorators: true` | Covers all widget source patterns |
| `module.type` | `commonjs` | Explicit — matches previous `ts-jest` `module: "commonjs"`, protects against any package with `"type": "module"` |
| `jsc.transform.react.runtime` | `automatic` | No `import React` needed in JSX files |

### Performance

| Metric | Before | After | Delta |
|---|---|---|---|
| Transform engine | ts-jest (tsc) | @swc/jest (Rust) | — |
| PWT own tests (6 suites, pure TS utilities) | 13.49s | 14.17s | ~flat (no TSX, no React) |
| Expected speedup for TSX-heavy consumers | baseline | **3–5×** | ~70–80% reduction |
| Type-checking in tests | yes (redundant) | no | eliminated |
| Test runner | jest-jasmine2 (deprecated) | jest-circus | ~10–15% additional |

PWT's own tests are pure TypeScript utilities with no TSX or React transforms, so they don't exercise the hot path where `@swc/jest` wins. Consumers with TSX-heavy test suites will see the 3–5× improvement.

All 79 existing PWT unit tests pass after the change.

### Downstream impact

All consumers of this package get the speedup automatically on the next version bump — zero changes required on their end.

### CODE REVIEW

**Review by iobuhov — APPROVED** [review]

File: packages/pluggable-widgets-tools/test-config/jest.config.js [hunk 1]
```diff
@@ -4,18 +4,23 @@ const projectDir = process.cwd();
 
 module.exports = {
     clearMocks: true,
-    testRunner: "jest-jasmine2",
+    testRunner: "jest-circus/runner",
+    testTimeout: 10000,
```
Comment (weirdwater) [inline packages/pluggable-widgets-tools/test-config/jest.config.js]: The description mentions that this allows locking async tests to fail faster, but the timeout is doubled. Is 5 seconds too short?

File: packages/pluggable-widgets-tools/test-config/jest.config.js [hunk 2]
```diff
@@ -4,18 +4,23 @@ const projectDir = process.cwd();
...
     transform: {
-        "^.+\\.tsx?$": [
-            "ts-jest",
+        "^.+\\.(t|j)sx?$": [
+            "@swc/jest",
             {
-                tsconfig: { module: "commonjs", target: "ES2019" }
+                jsc: {
+                    transform: { react: { runtime: "automatic" } },
+                    parser: { syntax: "typescript", tsx: true, decorators: true },
```
Comment (weirdwater) [inline packages/pluggable-widgets-tools/test-config/jest.config.js:17]: Does this care whether the code it receives is javascript?

File: packages/pluggable-widgets-tools/package.json
Comment (weirdwater) [inline packages/pluggable-widgets-tools/package.json:77]: We should now be able to remove jasmine2 and ts-jest here now, right?

**Review by weirdwater — CHANGES_REQUESTED** [review]
Tested it locally with the tests created by the generator. In these cases it works without issue.

File: packages/pluggable-widgets-tools/test-config/jest.config.js (hunk 2 above)
Comment (samuelreichert) [reply packages/pluggable-widgets-tools/test-config/jest.config.js:17]: No. @swc/jest with syntax: "typescript" parses JS fine. The unified pattern correctly replaces both old patterns (tsx? + jsx?).

Only edge case: if a .js file uses TypeScript-only syntax (type annotations, etc.) — but that's not valid JS anyway, so not a concern in practice.

File: packages/pluggable-widgets-tools/package.json
Comment (samuelreichert) [reply packages/pluggable-widgets-tools/package.json:77]: Still used by `jest.native.config.js` so we cant remove yet.

Comment (weirdwater) [reply packages/pluggable-widgets-tools/package.json:77]: Ah sad. Would have been nice to drop some dependencies as well

**Review by weirdwater — APPROVED** [review]

File: packages/pluggable-widgets-tools/CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Changed
 
 -   We changed the order of imports in generated widget prop types to match that of the eslint sort-imports rule.
+-   We replaced `ts-jest` with `@swc/jest` as the Jest transform (3–5× faster for TSX-heavy test suites) and switched the test runner from `jest-jasmine2` to `jest-circus`.
+
+### Breaking Changes
+
+-   The `jest-jasmine2` runner has been removed. Tests using Jasmine-specific globals (`jasmine.createSpy()`, `jasmine.objectContaining()`, etc.) will throw `ReferenceError: jasmine is not defined`. Replace with Jest equivalents: `jest.fn()`, `expect.objectContaining()`.
+-   Consumers who extended the base config with `globals['ts-jest']` options must migrate those settings to the `@swc/jest` transform config.
```
Comment (weirdwater) [inline packages/pluggable-widgets-tools/CHANGELOG.md]: I should have caught this sooner. The changelog changes need to be added to the `## Unreleased` section.

**Review by weirdwater — APPROVED** [review]

**Review by ancajuravle — APPROVED** [review]
